guide/popular-topics/permissions.md

```diff
@@ -0,0 +1,335 @@
+
+### Setting role permissions
+
+Permissions are set on roles, not the guild member itself. To change them you access the Role object (for example via `guild.roles`) and use the `.setPermissions()` method.
+Let's change the base permissions for @everyone so you can see it in action:
+
+```js
+// permission flag array
+guild.defaultRole.setPermissions(['SEND_MESSAGES', 'VIEW_CHANNEL']);
+
+// permission bit field in decimal representation
+guild.defaultRole.setPermissions(3072);
+```
+
+Any permission not referenced in the flag array or bit field are not granted to the role. 
+
+<tip>Note that flag names are literal. Although `VIEW_CHANNEL` grants access to view multiple channels the permission flag is still called `VIEW_CHANNEL` in singular.</tip>
+
+### Creating a role with permissions
+
+Alternatively you can provide permissions as a property of RoleData objects during role creation as an array of flag strings or a permission number:
+
+```js
+// permission bit field in decimal representation
+guild.createRole({ name: 'Mod', permissions: 8194 });
+
+// permission flag array
+guild.createRole({ name: 'Mod', permissions: ['MANAGE_MESSAGES', 'KICK_MEMBERS'] });
+```
+
+### Checking member permissions
+
+To know if a members roles have a permission enabled you can use the `.hasPermission()` method of the [GuildMember](https://discord.js.org/#/docs/main/stable/class/GuildMember?scrollTo=permissionsIn) class and provide a permission flag, array or number to check for. You can also specify if you want to allow the `ADMINISTRATOR` permission or the guild owner status to override this check with the following parameters.
+
+```js
+if (member.hasPermission('KICK_MEMBERS')) {
+	return console.log('This member can kick');
+}
+if (member.hasPermission(['KICK_MEMBERS', 'BAN_MEMBERS'])) {
+	return console.log('This member can kick and ban');
+}
+if (member.hasPermission('KICK_MEMBERS', false, false)) {
+	return console.log('This member can kick without allowing admin to override');
+}
+if (member.hasPermission('KICK_MEMBERS', false, false, false)) {
+	return console.log('This member can kick without allowing admin or owner to override');
+}
+```
+
+If you provide multiple permissions to the method it will only return `true` if all permissions you specified are granted.
+
+## Channel overwrites
+
+Permission overwrites control the abilities of members for this specific channel or a set of channels if applied to a category with synchronized child channels.
+
+As you have likely already seen in your desktop client, channel overwrites have three states: 
+
+- explicit allow `true` (green ✓)
+- explicit deny `false` (red X) 
+- default `null` (gray /)
+
+### Adding overwrites
+
+To add a permission overwrite for a role or user you access the channel object and use the `.overwritePermissions()` method. The first parameter is the target of the overwrite, either a Role or User object or their respective resolvable, the second is a PermissionOverwriteOptions object.
```

Comment:
Guild members have permissions, not users. d.js accepts a UserResolvable in this case, but it's quite the same thing.